src/org/labkey/test/util/CachingSupplier.java

```diff
@@ -0,0 +1,26 @@
+package org.labkey.test.util;
+
+import java.util.function.Supplier;
+
+public class CachingSupplier<T> implements Supplier<T>
```

Comment:
Add a comment that would help someone understand why / when they would want to use this class.